irc/client.go

```diff
@@ -184,6 +184,15 @@ func (client *Client) run() {
 	var line string
 	var msg ircmsg.IrcMessage
 
+	defer func() {
+		if r := recover(); r != nil {
+			client.server.logger.Error("internal", fmt.Sprintf("Client caused panic, disconnecting: %v", r))
+			debug.PrintStack()
```

Comment:
Hmm... I like this, but I think it makes more sense to log this whole string as an error -- makes digging into submitted debug logs a whole lot easier later. Could we do something along the lines of this?
```golang
client.server.logger.Error("internal", fmt.Sprintf("Client caused panic, disconnecting: %v\n%v", r, debug.Stack()))
```